src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
I was trying to implement the proposal I suggested, but unfortunately I don't think we can provide an usage as straightforward as I put it. By doing:

```scala
implicit def conv[T] = ConfigConvert.fromConfigFieldMapping[T](new WordDelimiterConfigFieldMapping(CamelCaseWordDelimiter, HyphenWordDelimiter))
```

We're effectively providing a `ConfigConvert` for all `T`, which is not possible nor something I think we would want to expose. The workaround I see for it is to have a specific type for `ConfigConverts` we derive implicitly and be able to customize those. However, that would expose an internal type via the API, which is something I'm also not comfortable with.

That being said, I think the current implementation offers the best compromise: it exposes a type that is meant to be used by `ConfigConverts` we derive implicitly and doesn't pollute user-defined `ConfigConverts`, as those don't need a `ConfigFieldMapping`.

@melrief and @leifwickland: What do you guys think?